File: src/main/java/be/vdab/scrumjava202504/products/ProductDetails.java
package be.vdab.scrumjava202504.products;

public class ProductDetails {
    private String productName;
    private long EAN;
    private String shelf;
    private int position;
    private int quantityOrdered;
    private double price;
    private int quantityStock;
    private String supplier;
    private double weight;

    public ProductDetails(String productName, long EAN, String shelf, int position, int quantityOrdered, double price, int quantityStock, String supplier, double weight) {
        this.productName = productName;
        this.EAN = EAN;
        this.shelf = shelf;
        this.position = position;
        this.quantityOrdered = quantityOrdered;
        this.price = price;
        this.quantityStock = quantityStock;
        this.supplier = supplier;
        this.weight = weight;
    }

    public long getEAN() {
        return EAN;
    }

    public String getShelf() {
        return shelf;
    }

    public int getPosition() {
        return position;
    }

    public int getQuantityOrdered() {
        return quantityOrdered;
    }

    public double getPrice() {
        return price;
    }

    public int getQuantityStock() {
        return quantityStock;
    }

    public String getProductName() {
        return productName;}

    public String getSupplier() {
        return supplier;
    }

    public double getWeight() {
        return weight;
    }
}

File: src/main/resources/static/js/productDetails.js
import {byId, capitalizeFirstWord} from "./util.js";

const orderData = JSON.parse(sessionStorage.getItem("orderData"));

if (!orderData) {
    byId("no-product-found").hidden = false;
}

try {
    const response = await fetch(`api/products/${orderData.id}/${orderData.shelf}/${orderData.position}`);

    if (response.ok) {
        const productDetails = await response.json();

        populateProductDetails(productDetails[0]); // Neem de eerste (en enige) productdetails
    } else {
        byId("error").hidden = false;
    }
} catch (error) {
    byId("error").hidden = false;
}

function populateProductDetails(product) {
    byId("name").innerText = capitalizeFirstWord(product.productName);
    byId("ean").innerText = product.ean;
    byId("rack").innerText = product.shelf;
    byId("place").innerText = product.position;
    byId("number-to-pick").innerText = product.quantityOrdered;
    byId("price").innerText = product.price.toFixed(2) + " €";
    byId("stock-available").innerText = product.quantityStock;
    byId("supplier").innerText = product.supplier;
    byId("weight").innerText = product.weight + " kg";
}

